components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -777,6 +836,15 @@ export const useSwap = () => {
         return 'insufficientLiquidity'
       }
 
+      // LiFi specific validations
+      if (quoteErrorUnion?.lifiError?.message) {
+        return quoteErrorUnion?.lifiError?.message?.includes(
+          'Could not find token'
+        )
+          ? 'insufficientLiquidity'
+          : 'unknownError'
```

Comment:
I wonder if this is better moved into core on the `LiFiError` object itself similar to `JupiterError` & `ZeroExError`? I wasn't able to find insufficient liquidity in a few quick tests but I _believe_ we can check for [`INSUFFICIENT_LIQUIDITY`](https://docs.li.fi/li.fi-api/li.fi-api/requesting-a-quote/possible-tool-errors) code, or move this string includes/contains check into core. Then we wouldn't have to update frontend if we added code support later.

Reply:
Awesome thanks! Updated iOS side in https://github.com/brave/brave-core/pull/22652/commits/0ed8b521eefc11b0bab7535a09aa7dc629c307b4 to check the code enum too.